evals/evaluation/deepeval/models/endpoint_models.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import abc
+import time
+from abc import ABC, abstractmethod
+from functools import cached_property
+from typing import List, Optional, Tuple, Union
+
+import requests
+from aiohttp import ClientSession, TCPConnector
+from deepeval.models.gpt_model import GPTModel
+from requests.exceptions import RequestException
+
+
+class TGIEndpointModel(GPTModel):
```

Comment:
TGI is specific, we can refer to it as 'OpeaServiceModel' as we may also use vLLM as a serving framework.

Reply:
different inference backends have different output format, so I define specific classes for different backends